### PR TITLE
Add documentation to OS provider

### PR DESCRIPTION
**Pull request by igchor**
and make it more OS-agnostic

### CODE REVIEW

File: include/umf/providers/provider_os_memory.h [hunk 1]
```diff
@@ -16,47 +16,56 @@ extern "C" {
...
 
     UMF_PROTECTION_MAX // must be the last one
 } umf_mem_protection_flags_t;
 
+/// @brief Visibility of the memory allocations
 typedef enum umf_mem_visibility_t {
-    UMF_VISIBILITY_SHARED,
-    UMF_VISIBILITY_PRIVATE,
+    UMF_VISIBILITY_SHARED, ///< Updates to the memory allocated using OS provider are visible to other processes.
+    /// TODO: need to expose functionality to share open the mapping in other process and explicit sync?
```
Comment (igchor) [inline include/umf/providers/provider_os_memory.h:32]: @vinser52 @pbalcer do we even want to support `UMF_VISIBILITY_SHARED` for OS provider?  I don't think we have a use case for this yet.

**Review by PatKamin — APPROVED** [review]

File: include/umf/providers/provider_os_memory.h (hunk 1 above)
Comment (vinser52) [reply include/umf/providers/provider_os_memory.h:32]: MPI mentioned such a use case if I remember correctly. The IPC mechanism is supposed to be used. BUt need to investigate deeply such use case

Comment (igchor) [reply include/umf/providers/provider_os_memory.h:32]: I've created to remove this param for now: https://github.com/oneapi-src/unified-memory-framework/pull/236 It should be fairly easy to restore if we find that it's needed

File: src/provider/provider_os_memory.c [hunk 2]
```diff
@@ -146,8 +146,16 @@ static hwloc_membind_policy_t translate_numa_mode(umf_numa_mode_t mode,
...
     case UMF_NUMA_MODE_BIND:
+        if (nodemaskEmpty) {
+            // nodeset must not be empty
+            return -1;
+        }
         return HWLOC_MEMBIND_BIND;
     case UMF_NUMA_MODE_INTERLEAVE:
+        if (nodemaskEmpty) {
+            // nodeset must not be empty
+            return -1;
```
Comment (bratpiorka) [inline src/provider/provider_os_memory.c:157]: could you add tests for this case?

Comment (igchor) [reply src/provider/provider_os_memory.c:157]: Done. I've extended the tests to cover all negative cases.

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**